PopPUNK/sketchlib.py

```diff
@@ -309,49 +311,65 @@ def constructDatabase(assemblyList, klist, sketch_size, oPrefix, estimated_lengt
             Output prefix for resulting sketch files
         estimated_length (int)
             Estimated length of genome, if not calculated from data
-        ignoreLengthOutliers (bool)
-            Whether to check for outlying genome lengths (and error
-            if found)
-            (default = False)
         threads (int)
-            Number of threads to use
-
-            (default = 1)
+            Number of threads to use (default = 1)
         overwrite (bool)
             Whether to overwrite sketch DBs, if they already exist.
-
             (default = False)
         reads (bool)
             If any reads are being used as input, do not run QC
-
             (default = False)
         strand_preserved (bool)
-            Ignore reverse complement k-mers
-
-            (default = False)
+            Ignore reverse complement k-mers (default = False)
         min_count (int)
             Minimum count of k-mer in reads to include
-
             (default = 0)
         use_exact (bool)
             Use exact count of k-mer appearance in reads
-
             (default = False)
+        qc_filter (string)
+            Behaviour on identifying sequences failing QC
+        retain_failures (bool)
+            Keep sketches of genomes that fail QC separate from main
+            database
+        length_sigma (int)
+            Number of SDs of length distribution beyond which sequences
+            are excluded
+        lower_length (int)
+            Threshold length below which sequences are excluded
+        upper_length (int)
+            Threshold length above which sequences are excluded
+        prop_n (float)
+            Proportion of ambiguous bases above which sequences are excluded
+        upper_n (int)
+            Number of ambiguous bases above which sequences are excluded
     """
+    # read file names
     names, sequences = readRfile(assemblyList)
-    if not reads:
-        genome_length, max_prob = assembly_qc(sequences, klist, ignoreLengthOutliers, estimated_length)
-        sys.stderr.write("Worst random match probability at " + str(min(klist)) +
-                            "-mers: " + "{:.2f}".format(max_prob) + "\n")
 
+    # create directory
     dbname = oPrefix + "/" + os.path.basename(oPrefix)
     dbfilename = dbname + ".h5"
     if os.path.isfile(dbfilename) and overwrite == True:
         sys.stderr.write("Overwriting db: " + dbfilename + "\n")
         os.remove(dbfilename)
 
+    # generate sketches
     pp_sketchlib.constructDatabase(dbname, names, sequences, klist, sketch_size,
                                    not strand_preserved, min_count, use_exact, threads)
+    
+    # QC sequences
+    if reads:
+        # will need to define reads QC function here
+        filtered_names = names
```

Comment:
I actually think you can remove this. reads still have counts of Ns and estimated assembly length (which can be estimated using the minhash), so should still work as below?

Reply:
`--reads` gone in f96c8d1.